sdk/nodejs/runtime/closure/createClosure.ts

```diff
@@ -880,7 +880,10 @@ async function getOrCreateEntryAsync(
             // Serialize functions recursively, and store them in a closure property.
             entry.function = await analyzeFunctionInfoAsync(obj, context, serialize, logInfo);
         }
-        else if (await isOutputAsync(obj)) {
+        else if (Output.isInstance(obj)) {
+            if (await (obj.isSecret || Promise.resolve(false))) {
+                throw new Error("Secret outputs may not be captured by a clsoure.");
```

Comment:
consider `cannot` instead of `may not` .

Reply:
:+1: